chia/rpc/full_node_rpc_api.py

```diff
@@ -32,17 +32,17 @@ def coin_record_dict_backwards_compat(coin_record: Dict[str, Any]) -> Dict[str,
     return coin_record
 
 
-async def get_nearest_transaction_block(blockchain: Blockchain, block: BlockRecord) -> Optional[BlockRecord]:
+async def get_nearest_transaction_block(blockchain: Blockchain, block: BlockRecord) -> BlockRecord:
     if block.is_transaction_block:
         return block
 
-    prev_hash = block.prev_transaction_block_hash
-    if prev_hash is None:
-        return None
+    prev_hash = blockchain.height_to_hash(block.prev_transaction_block_height)
```

Comment:
> instead of looking up the hash from the height, I think you should request the block with the given height directly from the DB (instead of `get_block_record_from_db()`).
> 
> But it sounds like this function will need to do a linear search to find the transaction block, right?

get off github and enjoy your vacation :)!

Reply:
Do you mean using `BlockStore.get_full_blocks_at`? - I'm not sure that makes much difference here. I'm not sure I understand the "linear search" comment, as the block as a back_pointer to the TX block in `prev_transaction_block_height`